### PR TITLE
Clean up unused code in Autolabel

### PR DESCRIPTION
**Pull request by rajasbansal**
# Pull Review Summary

## Description

Clean up unused code in Autolabel

## Type of change

- New feature (change which adds functionality)
- This change requires a documentation update

## Tests

an example dataset
refuel-cloud locally

### CODE REVIEW

File: src/autolabel/labeler.py [hunk 1]
```diff
@@ -168,40 +157,7 @@ def run(
...
-                    self.task_run,
-                    csv_file_name,
-                    gt_labels=dataset.gt_labels,
-                    additional_metrics=additional_metrics,
-                )
-            else:
-                self.task_run = self.db.create_task_run(
-                    csv_file_name, self.task_object.id, self.dataset_obj.id
-                )
+        self.all_annotations = []
```
Comment (yadavsahil197) [inline src/autolabel/labeler.py:160]: this shouldnt be an object property, lets keep the scope to this method only

File: src/autolabel/labeler.py [hunk 2]
```diff
@@ -80,15 +76,13 @@ def __init__(
         config: Union[AutolabelConfig, str, dict],
         cache: Optional[bool] = True,
         example_selector: Optional[BaseExampleSelector] = None,
-        create_task: Optional[bool] = False,
         console_output: Optional[bool] = True,
+        create_task: Optional[bool] = False,
```
[the remark this answers is not in the document]
Comment (rajasbansal) [reply src/autolabel/labeler.py:80]: we are not removing it as it is being used in refuel_cloud. will have to update that as well

File: src/autolabel/labeler.py (hunk 1 above)
Comment (rajasbansal) [reply src/autolabel/labeler.py:160]: makes sense!

File: src/autolabel/labeler.py [hunk 3]
```diff
@@ -795,25 +685,10 @@ def clear_cache(self, use_ttl: bool = True):
         self.transform_cache.clear(use_ttl=use_ttl)
 
     def save_annotation(self, annotation: LLMAnnotation, current_index: int, i: int):
```
[the remark this answers is not in the document]
Comment (rajasbansal) [reply src/autolabel/labeler.py:687]: yep same as above

File: src/autolabel/labeler.py [hunk 4]
```diff
@@ -138,11 +132,6 @@ def __init__(
         # Only used if we don't use task management
         self.all_annotations = []
```
[the remark this answers is not in the document]
Comment (rajasbansal) [reply src/autolabel/labeler.py:133]: same

File: src/autolabel/labeler.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (yadavsahil197) [reply src/autolabel/labeler.py]: we will fix it there

**Review by yadavsahil197 — APPROVED** [review]

**Pull request merged**